src/herder/SurgePricingUtils.cpp

```diff
@@ -389,14 +389,16 @@ SurgePricingPriorityQueue::canFitWithEviction(
             auto const& evictTxStack = *iter;
             auto const& evictTx = *evictTxStack->getTopTx();
             auto evictLane = mLaneConfig->getLane(evictTx);
-            auto evictOps = evictTxStack->getNumOperations();
             bool canEvict = false;
             // Check if it makes sense to evict the current top tx stack.
             //
             // Simple cases: generic lane txs can evict anything; same lane txs
             // can evict each other; any transaction can be evicted if per-lane
-            // limit hasn't been reached.
-            if (lane == GENERIC_LANE || lane == evictLane || neededLaneOps <= 0)
+            // limit hasn't been reached; any transaction can be evicted if
+            // there is not enough ops to evict from limited lane to fit the
+            // current transaction
+            if (lane == GENERIC_LANE || lane == evictLane ||
```

Comment:
I think this can be further simplified to
```
if (lane == GENERIC_LANE || lane == evictLane || neededTotalOps > neededLaneOps)
{
    canEvict = true;
}
else
{
    evictedDueToLaneLimit = true;
}
```
`neededLaneOps <= 0` check is redundant (0 is less than any non-0 value of total ops)
`neededTotalOps <= neededLaneOps` has to be true if we get into else branch.

Reply:
Indeed, fixed